vortex-array/src/canonical.rs

```diff
@@ -78,10 +78,14 @@ impl Canonical {
             Canonical::Struct(a) => struct_to_arrow(a)?,
             Canonical::VarBin(a) => varbin_to_arrow(a)?,
             Canonical::Extension(a) => {
-                if !is_temporal_ext_type(a.id()) {
-                    vortex_bail!("unsupported extension dtype with ID {}", a.id().as_ref())
+                if is_temporal_ext_type(a.id()) {
+                    temporal_to_arrow(TemporalArray::try_from(&a.into_array())?)?
+                } else {
+                    // Convert storage array directly into arrow.
+                    // NOTE: this loses the extension type information and we lose the ability to
+                    // round-trip back to Vortex.
+                    a.storage().into_canonical()?.into_arrow()?
```

Comment:
perhaps this was better just erroring, not sure

Reply:
I think this needs to delegate to the extension array, i.e. extension arrays need their own into_arrow method